src/Qir/Runtime/stdlib/src/arrays.rs

```diff
@@ -0,0 +1,335 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#![deny(clippy::all, clippy::pedantic)]
+
+use crate::update_counts;
+use std::{rc::Rc, usize};
+
+#[derive(Debug, Clone)]
+pub struct QirArray {
+    pub(crate) elem_size: usize,
+    pub(crate) data: Vec<u8>,
+}
+
+/// # Panics
+///
+/// This function panics if the passed in sizes do not fit into the usize type for the
+/// current platform.
+#[no_mangle]
+pub extern "C" fn __quantum__rt__array_create_1d(elem_size: u32, size: u64) -> *const QirArray {
+    let elem_size = elem_size.try_into().unwrap();
```

Comment:
Ah, that's true. I'd been working through the `unwrap()` usages in the code and replacing them with `expect()` and must not have gotten to this file. Incidentally, the use of `expect()` removes the clippy requirement for a `# Panics` docs section, even though the panic is still possible, presumably because the convention for `expect()` is that it is genuinely never expected to happen. Here that doesn't seem the case, because if we compiled and used this runtime in an environment where `usize` was smaller than `u64` it is possible runtime provided values would trigger the panic.